shipyard_airflow/control/dag_runs.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import falcon
+import requests
+
+from .base import BaseResource
+
+class DagRunResource(BaseResource):
+
+    authorized_roles = ['user']
+
+    def on_post(self, req, resp, dag_id, run_id=None, conf=None, execution_date=None):
+        req_url = 'http://localhost:32080/api/experimental/dags/{}/dag_runs'.format(dag_id)
+
+        response = requests.post(req_url,
+                                 json={
+                                 "run_id": run_id,
+                                 "conf": conf,
+                                 "execution_date": execution_date,
+                             })
+        if not response.ok:
+            raise IOError()
```

Comment:
I think we settled on a pattern for this, but I'm not sure if Anthony was on the call it was discussed. It will be something like

```
class BaseResource(object):
    def return_error(self, resp, status_code, message="", retry=False):
        """
        Write a error message body and throw a Falcon exception to trigger an HTTP status

        :param resp: Falcon response object to update
        :param status_code: Falcon status_code constant
        :param message: Optional error message to include in the body
        :param retry: Optional flag whether client should retry the operation. Can ignore if we rely solely on 4XX vs 5xx status codes
        """
        resp.body = json.dumps({'type': 'error', 'message': message, 'retry': retry})
        resp.content_type = 'application/json'
        resp.status = status_code

class FooResource(BaseResource):
    def on_get(self, req, resp):
        if req.content_type == 'text/plain':
            # Do stuff
        else:
            self.return_error(resp, falcon.HTTP_415, 'Resource requires body of type text/plain')
            return
```